tests/test_nagbot.py

```diff
@@ -24,6 +24,35 @@ def setup_instance(self, state: str, stop_after: str = '', terminate_after: str
                         contact='stephen',
                         nagbot_state='')
 
+    def test_taggable(self):
+        valid_date = '2019-01-01'
+        fully_tagged = self.setup_instance(stop_after=valid_date, terminate_after=valid_date)
+        warning_terminate = self.setup_instance(stop_after=valid_date,
+                                                terminate_after='Frigid froyo feeds Flo (Nagbot: Warned on 2019-01-01)')
+        unknown_stop = self.setup_instance(stop_after='Wonderful Willamette wombats', terminate_after=valid_date)
+        empty_stop = self.setup_instance(terminate_after=valid_date)
+        empty_terminate = self.setup_instance(stop_after=valid_date)
+        empty_stop_and_terminate = self.setup_instance()
+        unknown_terminate = self.setup_instance(stop_after=valid_date,
+                                                terminate_after='Pope Paul pleases proletariat people')
+
+        # These have something in stop_after and some kind of date for terminate_after
+        assert nagbot.is_taggable(fully_tagged) == False
+        assert nagbot.is_taggable(warning_terminate) == False
+        assert nagbot.is_taggable(unknown_stop) == False    # This one will get stopped due to the warning system
+
+        # stop_after should be filled in if it's not present
+        assert nagbot.is_taggable(empty_stop) == True
+
+        # terminate_after should be filled in if it's not present
+        assert nagbot.is_taggable(empty_terminate) == True
+
+        # stop_after and terminate_after should be filled in if they're not present
+        assert nagbot.is_taggable(empty_stop_and_terminate) == True
+
+        # terminate_after must have some kind of date in it
+        assert nagbot.is_taggable(unknown_terminate) == True
+
```

Comment:
[PEP8] Extra newline (it's two between functions but one between methods)

Reply:
Fixed